### PR TITLE
[Mint-X/Cinnamon] Cleanup and fixes

### PR DESCRIPTION
**Pull request by Odyseus**
## TL;DR

- Removed classes/IDs/properties that aren't used anymore by any version of Cinnamon that this theme supports.
-  Fixed missing styles, duplicated properties and wrong references in comments.

### Cleaning

- **.lg-*** and **#LookingGlassPropertyInspector**: Everything related to these classes and *ID* was removed from Cinnamon back in 2014 ([Cinnamon PR#3779](linuxmint/Cinnamon#3779)). **#LookingGlassDialog** was kept because it's still used by Melange. But the styling of elements *inside* this *ID* (**#Toolbar**, **StScrollBar**, etc.) were removed because it seems that they were also removed.
- **.end-session-dialog-***, **.hotplug-*** and **.polkit-dialog-***: Everything related to these classes was removed from Cinnamon back in 2013 ([Cinnamon commit where these classes were removed from the default theme](linuxmint/Cinnamon@b8d1dbb).
- **.popup-menu** and **.popup-menu-boxpointer**: These classes were left for retro-compatibility with Cinnamon versions under 3.2.x. But since version 3.4.x of Cinnamon, it is not possible to make this theme retro-compatible due to the removal of the programmatically added spacing on the **TextIconApplet** prototype by [Cinnamon PR#6271](linuxmint/Cinnamon#6271). I also checked the applets repository, and none of the applets are using these classes anymore.
- **.menu-category-button-button:hover**: This selector's class never existed. And even if it is a typo meant to style the **menu-category-button-selected** class on hover, its result is horrendous. So, I removed it.
- **#overview-corner**: The hot corner icons were removed by [Cinnamon PR#6423](linuxmint/Cinnamon#6423). The images **overview.png** and **overview-hover.png** were also removed.
- **.panel-button** and **.panel-menu**: This classes usage was removed in this [Cinnamon commit](linuxmint/Cinnamon@91b0b76).
    - The third-party xlet called **Weather** defines the **panel-button** class to be used for styling, [but never really uses it](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/weather%40mockturtl/files/weather%40mockturtl/applet.js#L123).
    - The third-party applet called **CPU Frequency Applet** uses the **panel-button** class [but then it removes it](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/cpufreq%40mtwebster/files/cpufreq%40mtwebster/applet.js#L150).
- **.panel-status-button**: Hard to pin-point were this class was removed. My guess is around [Cinnamon PR#4125](linuxmint/Cinnamon#4125). What's easy to realize is that the styles defined in this class are almost the same as the styling for the **applet-box** class. Additionally, the **-natural-hpadding** and **-minimum-hpadding** properties defined in this class are never programmatically applied.
    - Class used by the default applet called **Keyboard** (UUID keyboard@cinnamon.org). I think that this class should be removed from this applet.
- **.label-shadow**: This class is defined by the **TextShadower** prototype. This prototype hasn't been used in Cinnamon [since around 2012](linuxmint/Cinnamon@3f25240) and, if it is usefull to know, it was removed from Gnome Shell [in 2015](333becef45d58be20ea0b9d4404c2c382a9db117). I removed it because [of the comment in this prototype's definition](https://github.com/linuxmint/Cinnamon/blob/master/js/ui/panel.js#L1201), which suggests that the prototype might get removed in case of it not been used anymore (my guess).
- Removed `opacity` and `mouseover-opacity` properties from **#notification** selector. These properties were used before the **Have notifications fade out when hovered over** setting was introduced ([Cinnamon commit](linuxmint/Cinnamon@dae0a21). Now a days, neither of the properties are used programmatically nor the aforementioned setting exist.

### Fixes

- Added **url-highlighter** class styling: Links inside notifications weren't styled by the theme. They were hard-coded to **#ccccff**, which makes sense for the default theme with black backgrounds (the one called Cinnamon), but made these links practically unreadable on grey backgrounds like the Mint-X themes have.
- Added support for the **progress** class. Class used by the overlay that display a progress indicator on the window list items.
- Several entry selectors had the **selected-color** property declared twice.
- Several comments that separate sections have a reference to the file that define the classes that are being styled. Some of those comments were wrong (they made reference to the file names from before the *default applets were converted into applets* (somewhere around this [Cinnamon commit](linuxmint/Cinnamon@30d1c58). For the sake of completeness, I also added references to the sections that didn't have them.

### CODE REVIEW

**Comment by JosephMcc** [comment]
Is this ready to go?

**Comment by Odyseus** [comment]
Hello, @JosephMcc.

Yes, it is as ready as it can be. I had some *setbacks* with the **progress** class (it didn't look nice in all four panels). I did what I could.

I have more *fixes* that could be made to the theme. But they change the *desing concept* of the Mint-X theme in some way or another. So I will add them in individual PRs so they can be approved/rejected individually.

**Comment by JosephMcc** [comment]
Thanks. Dealing with the progress thing with the image based assets is a bit of a pain.
